app/grandchallenge/profiles/views.py

```diff
@@ -197,3 +206,94 @@ def form_invalid(self, form):
         # and display an error message
         messages.add_message(self.request, messages.ERROR, "Incorrect token.")
         return response
+
+
+class SubscriptionUpdate(SuccessMessageMixin, UserPassesTestMixin, UpdateView):
+    model = UserProfile
+    form_class = SubscriptionPreferenceForm
+    template_name = "profiles/subscription_form.html"
+    success_message = "You successfully updated your subscription preferences."
+    raise_exception = True
+
+    def test_func(self):
+        try:
+            token = self.validate_token()
+        except BadSignature:
+            return False
+
+        if (
+            not get_user_model()
+            .objects.filter(username__iexact=token, is_active=True)
+            .exists()
+        ):
+            return False
+
+        return True
+
+    def validate_token(self):
+        return Signer(salt=UNSUBSCRIBE_SALT).unsign(self.kwargs.get("token"))
```

Comment:
In `test_func` the result of this method is called `token` when it is returning a username. Might make the usage a little clearer?

```suggestion
    @property
    def username_from_token(self):
        return Signer(salt=UNSUBSCRIBE_SALT).unsign(self.kwargs.get("token"))
```